providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/operators/job.py

```diff
@@ -279,8 +290,6 @@ def on_kill(self) -> None:
             if self.termination_grace_period is not None:
                 kwargs.update(grace_period_seconds=self.termination_grace_period)
             self.job_client.delete_namespaced_job(**kwargs)
```

Comment:
Did you check the `on_kill` method after removing the pod's `on_kill`? Did the pod remove correctly?

Reply:
Regarding this: I ran the job operator and hit ```Clear Instance``` button. After this I received: Direct database access via the ORM is not allowed in Airflow 3.0.  And don´t see that on_kill method was called. I´ll try to pull recent changes and retest it.

Reply:
Ok,  I tried again, but unfortunately the method on_kill is not called neither on Clear Instance nor Mark task failed manually. I´ll investigate it tomorrow

Reply:
I managed to call the on_kill method.  It is not killing the pods, but I added the propagation_policy and default it to Background. After this Kubernetes will not only kill the job but all of its dependents too. Can we consider it fixed? @MaksYermak